main.tf

```diff
@@ -1,3 +1,10 @@
+locals {
+  instance_count = length(var.private_route_table_ids)
+  # split each subnet into two routes
+  interface_cidrs = flatten([for i in range(local.instance_count) : ["0.0.0.0/1", "128.0.0.0/1"]])
```

Comment:
This is something that I've been considering, hence my "future work" item:

> - Even greater scale could be achieved with larger instances and more ENIs. This could either be configurable or dynamic based on autoscaling.

But one of the issues here is that these EC2 instances have a maximum number of interfaces depending on instance type. So for more than just 2 ENIs, for example, we would simultaneously need to scale up to a 'large' or beyond.